app/Metadata/Geodecoder.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/** @noinspection PhpUndefinedClassInspection */
+
+namespace App\Metadata;
+
+use App\Configs;
+use App\Logs;
+use Cache;
+use Geocoder\Provider\Cache\ProviderCache;
+use Geocoder\Provider\Nominatim\Nominatim;
+use Geocoder\Query\ReverseQuery;
+use Geocoder\StatefulGeocoder;
+use GuzzleHttp\HandlerStack;
+use Spatie\GuzzleRateLimiterMiddleware\RateLimiterMiddleware;
+use Spatie\GuzzleRateLimiterMiddleware\Store;
+
+class Geodecoder
+{
+	/**
+	 * Get http provider with or without caching.
+	 *
+	 * @return mixed Geocoder Provider
+	 */
+	public static function getGeocoderProvider()
+	{
+		$stack = HandlerStack::create();
+		$stack->push(RateLimiterMiddleware::perSecond(1));
+
+		$httpClient = new \GuzzleHttp\Client([
+			'handler' => $stack,
+			'timeout' => Configs::get_value('location_decoding_timeout'),
+		]);
+
+		$httpAdapter = new \Http\Adapter\Guzzle6\Client($httpClient);
+
+		$cachedProvider = null;
+		// $caching_type = Configs::get_value('location_decoding_caching_type');
```

Comment:
I can't see this config, but there does seem to be language. Is this a `TODO` line?

Reply:
actually that was in the first implementation when there was multiple cache set-up. Not the case anymore.